Add hard/soft contrast options for gruvbox dark

ianpan870102 opened the pull request:
Added contrast options hard/soft for gruvbox dark, fixed documentation typo for gruvbox light, and tweaked a few other minor details (mostly colour improvements and naming refactors.).

ianpan870102 commented:
This PR fixes #470

hlissner commented on themes/doom-gruvbox-theme.el:
```diff
@@ -20,66 +20,93 @@ determine the exact padding."
+
+   ;; Standardized official colours from gruvbox
+   (grey       '("#928374" "#909090" "brightblack"))   ; gray
+   (gray       '("#928374" "#909090" "brightblack"))   ; gray
```
Can you remove `gray`? Not sure why we need a duplicate; it isn't even used.

ianpan870102 replied:
Sure, removed.

ianpan870102 commented:
Here are the before-and-after screenshots:

### Before
![before](https://user-images.githubusercontent.com/36686900/82774207-c19f6600-9e76-11ea-98f4-6d8c209ecf09.png)

### After
#### Gruvbox (dark) Hard
![hard](https://user-images.githubusercontent.com/36686900/82774230-d7149000-9e76-11ea-986d-bdbe8430f045.png)


#### Gruvbox (dark) Medium
![medium](https://user-images.githubusercontent.com/36686900/82774234-daa81700-9e76-11ea-878c-6f57647e6326.png)

#### Gruvbox (dark) Soft

![soft](https://user-images.githubusercontent.com/36686900/82774251-e267bb80-9e76-11ea-943c-7aeed5f38e01.png)

ianpan870102 commented:
P.s. I fixed-up my commits to keep commits cleaner.

ema2159 commented:
Hey Ian! Looks nice! I'll review them as soon as I find a little bit of time.

ianpan870102 commented:
Thanks man, take your time!
By the way @ema2159  I thought I'd give you a heads-up regarding the refactoring I made, it's because I noticed that the person who originally ported doom-gruvbox introduced a new colour olive that takes the colour of green, and green took the colour of aqua. This was not in-line with the colour definitions of the original gruvbox palette on the github page, and I had a tiny bit of hard time when tweaking the colours. So I decided to help standardise the naming, **without suffering existing declarations**:  those that were defined to be olive are now green, which is a more correct name, yet with the same HEX value. To improve readability for future changes, I've added comments after every HEX value, indicating its original name from the original gruvbox palette, e.g. `bright-aqua, bright-orange, bg0, bg0_h, bg0_s` etc.

On top of that, I've corrected the appearance of trivial UI stuff like line numbers, HTML/web-mode tag bracket face and comments colours.

ianpan870102 commented:
[Note:] This PR now not only fixes #470 but also https://github.com/hlissner/emacs-doom-themes/issues/458

ianpan870102 commented:
@ema2159 @hlissner friendly ping

ema2159 commented:
Hey Ian! It looks pretty nice! The only thing I'm a little bit concerned with is the thing with the standardized names. Although I think having standardized names is great, I'm seeing a lot of duplicate colors, so I would rather put a comment with the official spec name of the color at the right of the doom color name, and add an extra color only if it's not in the doom palette. Otherwise, the changes are great! I think for quite some time we've been missing the opportunity of using custom variables in order to expand the theme catalog even more.

ianpan870102 commented:
> I'm seeing a lot of duplicate colors, so I would rather put a comment with the official spec name of the color at the right of the doom color name, and add an extra color only if it's not in the doom palette.

@ema2159 I get your point. The reason I did so was because I find it sometimes more convenient and 'readable' to have two different variable names referencing to the same colour (i.e. same hex value). 
That said, I'll try to revert some of the **duplicate declarations** and **remove** all **unused** extra faces to make the definitions more concise.

ianpan870102 commented:
I've taken your suggestion into account in [this commit](https://github.com/ianpan870102/emacs-doom-themes/commit/5eb722aafe1ef3a4a0c2abcb62f05085472c4268), ready for your final review :-)

ema2159 commented:
Thanks Ian! Sorry for making you do double the work 😅. I'll merge it soon! Thanks for helping that much!

ianpan870102 commented:
> Thanks Ian! Sorry for making you do double the work

No problem at all, you're too kind. Happy to contribute!

hlissner commented:
@ianpan870102 Do you use `solaire-mode`? It seems `doom-gruvbox`'s `bg` and `bg-alt` are the wrong way around; with `bg-alt` being brighter than `bg`. Was that intentional?

What I expected:
![image](https://user-images.githubusercontent.com/510883/90461471-5ad7c480-e0d4-11ea-8528-100c330bfcbd.png)

What I see:
![image](https://user-images.githubusercontent.com/510883/90461476-5dd2b500-e0d4-11ea-8f74-2f486e3d77a5.png)

hlissner commented:
@ianpan870102 Sorry to cram two questions in one, but I have another one:

A user told me that the color for `which-key-group-description-face` is too low-contrast; difficult to distinguish from other keys. Would you mind if I changed it? And if so, what would you suggest it be changed to? I'm thinking yellow, but I'm not familiar enough with the theme.

ianpan870102 commented:
> It seems doom-gruvbox's bg and bg-alt are the wrong way around; with bg-alt being brighter than bg. Was that intentional?

Hi @hlissner, although I'm not a solaire user, I did take into account the `bg` and `bg-alt` faces when modifying the theme. **It was indeed my intention to make the secondary `bg-alt` appear brighter than the default `bg`**. However, I'm willing to file another PR and make `bg-alt` darker if it better adheres to `doom-themes` design guidelines for the sake of consistency. Let me know what you think!

> Would you mind if I changed it? And if so, what would you suggest it be changed to? I'm thinking yellow

By all means go ahead :) Yellow is a good choice indeed.
